homeassistant/components/sensor/apcupsd.py

```diff
@@ -0,0 +1,107 @@
+"""
+homeassistant.components.sensor.apcupsd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provides a sensor to track various status aspects of a UPS.
+"""
+import logging
+
+from homeassistant.core import JobPriority
+from homeassistant.const import TEMP_CELCIUS
+from homeassistant.helpers.entity import Entity
+from homeassistant.components import apcupsd
+
+
+DEPENDENCIES = [apcupsd.DOMAIN]
+
+DEFAULT_NAME = "UPS Status"
+
+SPECIFIC_UNITS = {
+    "ITEMP": TEMP_CELCIUS
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """
+    Ensure that the 'type' config value has been set and use a specific unit
+    of measurement if required.
+    """
+    typ = config.get(apcupsd.CONF_TYPE)
+    if typ is None:
+        _LOGGER.error(
+            "You must include a '%s' when configuring an APCUPSd sensor.",
+            apcupsd.CONF_TYPE)
+        return
+    typ = typ.upper()
+
+    # Get a status reading from APCUPSd and check whether the user provided
+    # 'type' is present in the output. If we're not able to check, then assume
+    # the user knows what they're doing.
+    # pylint: disable=broad-except
+    status = None
+    try:
+        status = apcupsd.GET_STATUS()
+        if typ not in status:
+            _LOGGER.error(
+                "Specified '%s' of '%s' does not appear in the APCUPSd status "
+                "output.", apcupsd.CONF_TYPE, typ)
+            return
+    except Exception as exc:
```

Comment:
What's with the broad exception? Must be able to be more specific.

However, this can all be moved to your Data class.

Also, make sure you do not continue setting up if on error: `return False`